esmvalcore/_recipe/check.py

```diff
@@ -392,7 +392,10 @@ def _check_format_years(date):
 
 def _check_timerange_values(date, timerange):
     try:
-        isodate.parse_date(date)
+        if "T" in date:
+            isodate.parse_datetime(date)
+        else:
+            isodate.parse_date(date)
     except ValueError:
         try:
             isodate.parse_duration(date)
```

Comment:
In other parts of the code we use the following syntax:

https://github.com/ESMValGroup/ESMValCore/blob/99c6a7be1e995f8b4fcb61f3a44bd46aa828c397/esmvalcore/preprocessor/_time.py#L121-L130

I am honestly not sure which one is better. Will all datetimes contain a "T"? If not, then I would go for this other option.

Reply:
that's a very good call, Manu! I *think* we should change it there too with a check on T and call the correct isodate function. `parse_date()` will fail if there is a T in it, and `parse_datetime()` will fail if there is no T in it (ah Mr T :grin: ) - or, we can let it fail so we prompt the user to change their date(time) strings in the recipe, though this would be very annoying if those are actually correct - what do you think, you and @bouweandela ?

Reply:
> Will all datetimes contain a "T"? 

Sorry, forgot to answer: no, some are just eg `19990605` ie just YYYYMMDD

Reply:
damn am verbose today! Note that `except isodate.isoerror.ISO8601Error:` is actually a bit too general and may fail now if the string is eg `YYYYMMDDTxxHxxMxxS`

Reply:
Yeah, agreed, just like python's `datetime.datetime` also handles date with and without time. But if the existence of "T" is enough to distinguish the two, we should implement your solution. And maybe include a test for "P" to check if a duration is given?

Reply:
> damn am verbose today! Note that `except isodate.isoerror.ISO8601Error:` is actually a bit too general and may fail now if the string is eg `YYYYMMDDTxxHxxMxxS`

But that's not a valid format, is it? See https://github.com/ESMValGroup/ESMValCore/issues/2545. So I guess removing the try/except and just checking for the existence of "P" and "T" in the string is the best solution?

Reply:
yeah that's exactly what I think too! Lemme massage that then :beer:

Reply:
@schlunma @bouweandela have a look at cc94a26f0c02d03a74e42d78b9692c3ec1c020ff and tell me if you think it's a good framework - I'll make similar changes to `_time.py` and `local.py` (whatever that is) :beer:

Reply:
I like it! Maybe we could even put that if/elif/else construct in a little function and use that everywhere? 😁

Reply:
sorry, kept editing the commit hash coz I keep forgetting I gotta Ruff it :grin: Well, we could, but that might be overkill coz we have to allow for one or two checked strings etc

Reply:
ah crap! Both P and T could be in the same string and that's still duration, need to refactor. Off home though, in the morrow :pizza:

Reply:
ah done that, now am off to :pizza: